core/stat/base_node.go

```diff
@@ -44,6 +44,23 @@ func NewBaseStatNode(sampleCount uint32, intervalInMs uint32) *BaseStatNode {
 	}
 }
 
+// NewCustomizedBaseStatNode creates the BaseStatNode with specified sampleCount
+// and intervalInMs.
+func NewCustomizedBaseStatNode(sampleCount uint32, intervalInMs uint32) (*BaseStatNode, error) {
```

Comment:
NewCustomizedBaseStatNode  it is not called and can be implemented in your own code

Reply:
But the fields of BaseStatNode are unexported, how do I implement it with my own code? 😢

Reply:
I didn't notice this branch, you can do that